packages/sacado/src/new_design/Sacado_Fad_Exp_GeneralFadTraits.hpp

```diff
@@ -215,7 +215,19 @@ namespace Teuchos {
 
 // KokkosComm
 #if defined(HAVE_SACADO_KOKKOSCORE) && defined(HAVE_SACADO_TEUCHOSKOKKOSCOMM) && defined(HAVE_SACADO_VIEW_SPEC) && !defined(SACADO_DISABLE_FAD_VIEW_SPEC)
+
+// We are hooking into Kokkos Core internals here
+// Need to define this macro since we include non-public headers
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_CORE
+#endif
 #include "KokkosExp_View_Fad.hpp"
+#ifdef KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_CORE
+#undef KOKKOS_IMPL_PUBLIC_INCLUDE
+#undef KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_CORE
+#endif
+
```

Comment:
Does this mean every user that includes KokkosExp_View_Fad.hpp needs to define these first?  That could be a problem.  I don't understand why it is necessary though since these are defined inside KokkosExp_View_Fad.hpp itself.

Reply:
@etphipp no, these guards are to allow for private Kokkos header usage within Sacado and Stokhos. It not needed and does not affect codes that include `KokkosExp_View_Fad.hpp` and such

Reply:
I don't understand then why those macros are needed here, but not when a user includes KokkosExp_View_Fad.hpp.

Reply:
Without the guards Kokkos will warn with the current 3.7.00 snapshot; with Kokkos develop branch (post-3.7.00) the deprecation warning is updated to a static_assert and code will not compile

Reply:
I understand that.  What I do not understand is why those macros are required when including KokkosExp_View_Fad.hpp here, but not in user code.

Reply:
The users are just not allowed to include `Kokkos_Layouts.hpp`

Reply:
This file does not include Kokkos_Layouts.hpp.  It just includes KokkosExp_View_Fad.hpp, which does include Kokkos_Layouts.hpp, but defines these exact same macros beforehand.  So first, I don't understand why they need to be defined here as well before the include of KokkosExp_View_Fad.hpp, and second, if they must be defined here, why they magically don't need to be defined if a user includes KokkosExp_View_Fad.hpp.

Reply:
Yes, they should not be needed here

Reply:
Agreed this should be reverted, I bet Nathan added those here before adding them in KokkosExp_View_Fad where they are needed

Reply:
Yep, Christian is right. Sorry for the mixup @etphipp I made the changes to remove from the file and rebuilding to confirm compilation then I'll push.
Github didn't show the line with the include line for `#include "KokkosExp_View_Fad.hpp"` in the commented highlights, I assumed we were talking about something like `#include "Kokkos_View.hpp"`

Reply:
@crtrott @etphipp fixed with cd7ac69ddbb0fe4e0d3ed3cdaa698fd943706328

Reply:
No worries, thanks!